staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -1629,7 +1660,7 @@ func (r Result) Error() error {
 	// to be backwards compatible with old servers that do not return a version, default to "v1"
 	out, _, err := r.decoder.Decode(r.body, &schema.GroupVersionKind{Version: "v1"}, nil)
 	if err != nil {
-		klog.V(5).Infof("body was not decodable (unable to check for Status): %v", err)
+		r.logger.V(5).Info("Body was not decodable (unable to check for Status)", "err", err)
```

Comment:
are we certain we'll never get a request here that got set up with a nil `logger`? does that need a guard?

Reply:
`r.logger` is not a pointer and the null struct is safe to use, therefore this [won't crash](https://github.com/go-logr/logr/blob/b8ba1898daddefb0d2a567a5b4bc420289722b9b/logr.go#L309-L311). Instead, [nothing gets logged](https://github.com/go-logr/logr/blob/b8ba1898daddefb0d2a567a5b4bc420289722b9b/logr.go#L273-L275).

To be absolutely sure that this log line gets emitted, `logger` would have to be turned into a pointer, with a fallback to `klog.Background()` if nil. But that then comes with the risk that adding a log call elsewhere forgets the if check and *that* then would crash.

There's also some additional memory overhead when using a pointer.

Overall I prefer to keep it as it is now.

Reply:
sounds good, I didn't know it was a nil-safe non-pointer